cpp/src/io/parquet/page_string_utils.cuh

```diff
@@ -98,21 +100,40 @@ __device__ inline void block_excl_sum(size_type* arr, size_type length, size_typ
   }
 }
 
-template <int block_size, bool has_lists>
-__device__ inline void convert_small_string_lengths_to_offsets(page_state_s* s)
+/**
+ * @brief Converts string sizes to offsets if this is not a large string column. Otherwise,
+ * atomically update the initial string offset to be used during large string column construction
+ */
+template <int block_size>
+__device__ void convert_string_lengths_to_offsets(page_state_s const* const state,
```

Comment:
How about something like `compute_page_or_initial_offsets`? I find the current name a little confusing/misleading.

Reply:
I agree that we need a better name for this function. I really don't like a function doing two different things but splitting stuff into functions just adds repetitions.

Reply:
Separated functionality into two separate functions to avoid future confusions 🙂